app/src/main/java/com/github/leuludyha/ibdb/di/AuthenticationModule.kt

```diff
@@ -6,26 +6,39 @@ import com.github.leuludyha.domain.model.user.MainUser
 import com.github.leuludyha.domain.model.user.preferences.UserStatistics
 import com.github.leuludyha.domain.repository.LibraryRepository
 import com.github.leuludyha.domain.useCase.GetAllWorkPrefsLocallyUseCase
-import com.google.firebase.auth.ktx.auth
-import com.google.firebase.ktx.Firebase
+import com.github.leuludyha.ibdb.util.Constant
 import dagger.Module
 import dagger.Provides
 import dagger.hilt.InstallIn
 import dagger.hilt.components.SingletonComponent
 import kotlinx.coroutines.flow.map
+import javax.inject.Singleton
 
 @Module
 @InstallIn(SingletonComponent::class)
 object AuthenticationModule {
 
+    @Singleton
     @Provides
     fun provideAuthenticationContext(libraryRepository: LibraryRepository): AuthenticationContext {
         return AuthenticationContext(
             MainUser(
-                Firebase.auth.currentUser?.uid!!,
-                Firebase.auth.currentUser?.displayName ?: "username",
-                Firebase.auth.currentUser?.photoUrl.toString(),
-                Firebase.auth.currentUser?.phoneNumber,
+                //TODO: Unfortunately cannot call Firebase directly from here (would fail if there was
```

Comment:
As we discuss, that is unfortunate but we haven't found a cleaner solution yet...
`Constants.USER_NOT_FOUND` is nice tho..